hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -154,21 +151,11 @@ def _compute_due_hours(self):
                 att_day.due_hours = 0
                 continue
 
-            # Contract
-            # sum the due hours extracted from the employee contract. The
-            # expected attendances are selected in update_calendar_attendance
-            # (further down in the file) which is called when an attendance
-            # day is created.
             due_hours = sum(att_day.mapped('cal_att_ids.due_hours'))
             # Leaves
-            if att_day.leave_ids:
-                for leave in att_day.leave_ids.filtered(
-                        lambda l: l.state in ('validate', 'validate1')):
-                    if not leave.holiday_status_id.remove_from_due_hours:
-                        working_day = self.env[
-                            'base.config.settings'].get_work_day_duration()
-                        due_hours -= leave.compute_work_day(
-                            att_day.date) * working_day
+            for leave in att_day.leave_ids.filtered(
+                    lambda l: l.state == 'validate' and not l.keep_due_hours):
+                due_hours -= leave.compute_leave_time(att_day.date)
```

Comment:
compute_leave_time is in range [0-24]. Are you sure that you want to deduce 24 hours of work here? 
In particular, if I leave at noon, leave time will return 12 but due hours should probably be reduced only by half. no?

Reply:
We discussed with David and choose to let it like that. This scenario will almost never happen and if it happens, the only consequence is the employee will have a bit more extra hours. The manager can correct this manually if needed (or can catch it before approving the leave request).

Reply:
Yup sounds good to me then !